### PR TITLE
Update href prop; restrict to https:// URL

### PR DESCRIPTION
**Pull request by blackfalcon**
# Alaska Airlines Pull Request

## Summary:

The scope of this update is to remove the ability to directly inject a string into the `href` attribute of the component. The string will be converted into a URL object and evaluated prior to the object's `href` being injected into the `href` attribute of the `<a>` element. 

This update's use is currently restricted to the auro-hyperlink. The functionality is available to the ods-hyperlink and can be updated later if desired.

## Type of change:

Please delete options that are not relevant.

- [x] New capability 
- [x] Revision of an existing capability
- [ ] Infrastructure change (automation, etc.)
- [ ] Other (please elaborate) 

## Checklist:

- [x] My update follows the CONTRIBUTING guidelines of this project
- [x] I have performed a self-review of my own update

**By submitting this Pull Request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

_Pull Requests will be evaluated by their quality of update and whether it is consistent with the goals and values of this project. Any submission is to be considered a conversation between the submitter and the maintainers of this project and may require changes to your submission._ 

**Thank you for your submission!**<br>
-- Orion Design System Team

### CODE REVIEW

File: src/component-base.js [hunk 1]
```diff
@@ -37,6 +37,22 @@ export default class ComponentBase extends LitElement {
     };
   }
 
+  safeUrl(href) {
+    if(href != undefined) {
+      const url = new URL(href, 'https://www.alaskaair.com')
```
Comment (geoffrey1218) [inline src/component-base.js]: Missing semicolons throughout function.

File: src/component-base.js [hunk 2]
```diff
@@ -37,6 +37,22 @@ export default class ComponentBase extends LitElement {
...
+
+      if(url.protocol === 'javascript:') {
+        return undefined
+      }
+
+      url.protocol = 'https:'
+      return url.href;}
```
Comment (geoffrey1218) [inline src/component-base.js]: Bracket should be on a separate line for consistency

Comment (geoffrey1218) [inline src/component-base.js]: Bracket should be on a separate line for consistency

File: demo/index.html [hunk 3]
```diff
@@ -62,6 +62,16 @@ <h3>auro-hyperlink examples</h3>
           </template>
       </demo-snippet>
 
+      <p>auro-hyperlink only accepts urls and will convert them to https:// protocol. </p>
+      <demo-snippet>
+        <template>
+            <p>See more about <auro-hyperlink href="/about">Alaska Airlines</auro-hyperlink> in this link. Link retruned in the DOM with full path to AS.com and https:// protocol.</p>
```
Comment (geoffrey1218) [inline demo/index.html]: Typo: "retruned" (also on next three lines)

File: src/component-base.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (blackfalcon) [reply src/component-base.js]: It appears to be addressed with this reference in Babel.js to a PR on core-js https://github.com/babel/babel/issues/7300

**Pull request merged**